lib/ajv.js

```diff
@@ -259,8 +259,17 @@ function Ajv(opts) {
 
 
     function addInitialSchemas() {
-        if (self.opts.meta !== false)
-            addMetaSchema(require('./refs/json-schema-draft-04.json'), META_SCHEMA_ID, true);
+        if (self.opts.meta !== false) {
+            var metaSchema = require('./refs/json-schema-draft-04.json');
+
+            addMetaSchema(metaSchema, META_SCHEMA_ID, true);
+
+            // Add other schemas according to JSON specification:
+            // http://json-schema.org/latest/json-schema-core.html
+            addMetaSchema(metaSchema, 'http://json-schema.org/schema', true);
+            addMetaSchema(metaSchema, 'http://json-schema.org/hyper-schema', true);
```

Comment:
hyper meta-schema is a different from meta-schema - it has additional keywords (and they are not implemented). I suggest not adding it, because it will increase the size of browser bundle. It can be manually added using addMetaSchema by users when needed.